packages/eslint-plugin/src/rules/only-throw-error.ts

```diff
@@ -65,13 +78,74 @@ export default createRule<Options, MessageIds>({
   defaultOptions: [
     {
       allow: [],
+      allowRethrowing: true,
       allowThrowingAny: true,
       allowThrowingUnknown: true,
     },
   ],
   create(context, [options]) {
     const services = getParserServices(context);
     const allow = options.allow;
+
+    function isRethrownError(node: TSESTree.Node): boolean {
+      if (node.type !== AST_NODE_TYPES.Identifier) {
+        return false;
+      }
+
+      const scope = context.sourceCode.getScope(node);
+
+      const smVariable = nullThrows(
+        findVariable(scope, node),
+        `Variable ${node.name} should exist in scope manager`,
+      );
+
+      const variableDefinitions = smVariable.defs.filter(
+        def => def.isVariableDefinition,
+      );
+      if (variableDefinitions.length !== 1) {
+        return false;
+      }
+      const def = smVariable.defs[0];
+
+      // try { /* ... */ } catch (x) { throw x; }
+      if (def.node.type === AST_NODE_TYPES.CatchClause) {
+        return true;
+      }
+
+      // promise.catch(x => { throw x; })
+      // promise.then(onFulfilled, x => { throw x; })
+      if (
+        def.node.type === AST_NODE_TYPES.ArrowFunctionExpression &&
+        def.node.params.length >= 1 &&
+        def.node.params[0] === def.name &&
+        def.node.parent.type === AST_NODE_TYPES.CallExpression
+      ) {
+        const callExpression = def.node.parent;
+
+        const parsedPromiseHandlingCall =
+          parseCatchCall(callExpression, context) ??
+          parseThenCall(callExpression, context);
```

Comment:
[Refactor] Both this rule and `no-floating-promises` run `parseCatchCall(node, context) ?? parseThenCall(node, context)`. Nowhere else uses either. Is there a reason to expose both functions?

```suggestion
          parsePromiseHandlingCall(callExpression, context);
```

Reply:
IMHO - yes. It's "coincidental" in a sense that these functions are both used essentially as `(parseThenCall() ?? parseCatchCall())?.onRejected` in the two rules they're used in. But you could imagine even if we just wanted to differentiate the error message based on whether it's a `.then()` or a `.catch()` we'd need these exposed separately, let alone if we need separate logic between the methods in a rule at some point. Slightly speculative I suppose but not really a stretch I don't think.

Reply:
Makes sense to me, thanks. Resolving as wontfix 🔪